pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java

```diff
@@ -80,9 +81,11 @@ public boolean isMatch(int docId) {
       return false;
     }
     valueIterator.skipTo(docId);
-    _numEntriesScanned++;
     int length = valueIterator.nextIntVal(intArray);
-    return evaluator.applyMV(intArray, length);
+    MutableInt numEntriesScannedInEvaluator = new MutableInt(0);
+    boolean ret = evaluator.applyMV(intArray, length, numEntriesScannedInEvaluator);
+    _numEntriesScanned += numEntriesScannedInEvaluator.toInteger();
```

Comment:
You can make _numEntriesScanned a MutableInt and keep passing it to the evaluators. In the getNumEntriesScanned(), return _numEntriesScanned.intValue()

Reply:
Please also fix the tests

Reply:
+1